### PR TITLE
JBPM-5493 - Validation falsely fails when adhoc subprocess has no out…

### PR DESCRIPTION
**Pull request by tsurdilo**
…going connection in a case definition

### CODE REVIEW

**Comment by tsurdilo** [comment]
test failure on org.jbpm.executor.impl.jms.JmsAvaiableJobExecutorTest.testAsyncAuditProducerPrioritizedJobs not related to changes in this pr

**Comment by tsurdilo** [comment]
jenkins retest this

**Comment by mswiderski** [comment]
@tsurdilo maybe we should put the check as well that the dynamic node is in dynamic process as well when omitting these checks? It should still enforce these rules when it is in regular process (not ad hoc) wdyt?

**Comment by ibek** [comment]
+1 for keeping the checks in processes

**Comment by tsurdilo** [comment]
@mswiderski @ibek - agree too :) will amend

**Comment by tsurdilo** [comment]
@mswiderski @ibek amended. thanks!

File: jbpm-flow/src/main/java/org/jbpm/ruleflow/core/validation/RuleFlowProcessValidator.java [hunk 1]
```diff
@@ -355,12 +355,16 @@ else if (node instanceof SubProcessNode) {
...
             } else if (node instanceof DynamicNode) {
                 final DynamicNode dynamicNode = (DynamicNode) node;
-                if (dynamicNode.getDefaultIncomingConnections().size() == 0) {
-                    addErrorMessage(process, node, errors, "Dynamic has no incoming connection");
-                }
-                if (dynamicNode.getDefaultOutgoingConnections().size() == 0) {
-                    addErrorMessage(process, node, errors, "Dynamic has no outgoing connection");
+
+                if(!process.isDynamic()) {
+                    if (dynamicNode.getDefaultIncomingConnections().size() == 0) {
```
Comment (ibek) [inline jbpm-flow/src/main/java/org/jbpm/ruleflow/core/validation/RuleFlowProcessValidator.java]: I noticed that the previous checks use acceptsNoIncomingConnections(dynamicNode) so we might use that one.

**Comment by tsurdilo** [comment]
@ibek ok :) done

**Comment by ibek** [comment]
@tsurdilo excellent, thank you, +1 for merge

**Pull request merged**